controllers/build/job/manager_test.go

```diff
@@ -177,22 +187,6 @@ var _ = Describe("JobManager", func() {
 			).To(
 				Equal(build.Result{Requeue: true, Status: build.StatusCreated}),
 			)
-
-			newJob := batchv1.Job{}
-
-			key := types.NamespacedName{Name: jobName, Namespace: namespace}
-
-			Expect(
-				client.Get(ctx, key, &newJob),
-			).To(
-				Succeed(),
-			)
-
-			Expect(
-				cmp.Diff(j, newJob),
-			).To(
-				BeEmpty(),
-			)
```

Comment:
These assertions have blown my mind. If I'm not wrong, it's checking what the `fake.Client` would return if we call the `Get()` method after creating the job. 
If that assumption is correct, I would say that's not the way to go. As a unit test, it should only rely on the returned result instead of the internal state of the dependencies.
WDYT?

Reply:
before your change, the unit test was based on the faked client, which is an actual client backed by a very simple object store. So this assertions were actually check what was actually stored in the object store. since we are moving to the real mocks, these assertions become obsolete